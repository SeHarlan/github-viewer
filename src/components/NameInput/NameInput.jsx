import React from 'react';
import PropTypes from 'prop-types';

const NameInput = ({ onButtonClick, onChange, username }) => (
  <>
    <button onClick={onButtonClick}>Submit</button>
    <input type="text" name="username" value={username} onChange={onChange} />
  </>
);

NameInput.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired
};



export default NameInput;
