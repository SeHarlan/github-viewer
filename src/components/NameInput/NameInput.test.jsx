import React from 'react';
import { shallow } from 'enzyme';
import NameInput from './NameInput';

describe('NameInput component', () => {
  it('renders NameInput', () => {
    const wrapper = shallow(<NameInput onButtonClick={() => { }} onChange={() => { }} username="Scott" />);
    expect(wrapper).toMatchSnapshot();
  });
});
