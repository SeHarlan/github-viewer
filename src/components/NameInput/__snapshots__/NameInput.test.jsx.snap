// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameInput component renders NameInput 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
  <input
    name="username"
    onChange={[Function]}
    type="text"
    value="Scott"
  />
</Fragment>
`;
